### PR TITLE
fix: issue with unsafe to chain command that is not an action

### LINKED ISSUE #140: cy.focused().type("text") shows cypress/unsafe-to-chain-command error
After an update to 2.14.0, the  cypress/unsafe-to-chain-command rule is triggered by code as
`cy.focused().type("text")`.
This should be no problem however, as the cypress docu states `cy.focused() is a query, and it is safe to chain further commands` (https://docs.cypress.io/api/commands/focused)

### PR DESCRIPTION
**Pull request by gkorakas-eli**
On version v2.14.0 it seems there is a change in the **unsafe-to-chain-command** rule that causes the following issue:

https://github.com/cypress-io/eslint-plugin-cypress/issues/140

This is probably a quick fix on this without changing the Regex used by additional methods. The test covers the case where there is a command that is not considered an action for Cypress.

- closes #140

### CODE REVIEW

File: lib/rules/unsafe-to-chain-command.js [hunk 1]
```diff
@@ -129,7 +129,7 @@ const isRootCypress = (node) => {
  */
 const isActionUnsafeToChain = (node, additionalMethods = []) => {
   const unsafeActionsRegex = new RegExp([
-    ...unsafeToChainActions,
+    ...unsafeToChainActions.map((action) => `^${action}$`),
```
Comment (dominicfraser) [inline lib/rules/unsafe-to-chain-command.js:132]: @gkorakas-eli This is nicer than the very specific hotfix I put forwards in https://github.com/cypress-io/eslint-plugin-cypress/pull/142 - I missed that this PR was open when making that PR to illustrate the problem.

If you pull in main now and revert the change I made in this file, but leave the tests I added, this looks like a better approach to me 👍 

WDYT @nagash77?

**Review by dominicfraser — APPROVED** [review]

File: lib/rules/unsafe-to-chain-command.js (hunk 1 above)
Comment (dominicfraser) [reply lib/rules/unsafe-to-chain-command.js:132]: LGTM - what do you think @nagash77?

Comment (dominicfraser) [reply lib/rules/unsafe-to-chain-command.js:132]: cc: @passbyval

Comment (passbyval) [reply lib/rules/unsafe-to-chain-command.js:132]: LGTM!

**Comment by imaliouris** [comment]
any update on this?

**Review by imaliouris — APPROVED** [review]

**Comment by dqiubread** [comment]
Any update on when this will be merged and a new release cut?

**Comment by MikeMcC399** [comment]
@gkorakas-eli 

Are you interesting in reviving this PR? It seems that it is still relevant.

**Comment by MikeMcC399** [comment]
@gkorakas-eli 

- I am looking back into the history and it seems that issue https://github.com/cypress-io/eslint-plugin-cypress/issues/140 was already solved by PR https://github.com/cypress-io/eslint-plugin-cypress/pull/144 and that was released in [2.15.1](https://github.com/cypress-io/eslint-plugin-cypress/releases/tag/v2.15.1)
- The author of PR https://github.com/cypress-io/eslint-plugin-cypress/pull/144 @dominicfraser wrote in https://github.com/cypress-io/eslint-plugin-cypress/pull/141/files#r1329258120 however that he thought your solution was better.
- Could you provide a test case which fails in [2.15.1](https://github.com/cypress-io/eslint-plugin-cypress/releases/tag/v2.15.1)  (or in the current version) and passes with your pending PR? I'm currently unsure what the exact issue is which this PR is intended to solve.

**Comment by gkorakas-eli** [comment]
> @gkorakas-eli
> 
> * I am looking back into the history and it seems that issue [cy.focused().type("text") shows cypress/unsafe-to-chain-command error #140](https://github.com/cypress-io/eslint-plugin-cypress/issues/140) was already solved by PR [fix: unsafe-to-chain-command: Fix the false positive of 'focus' regex matching 'focused' #144](https://github.com/cypress-io/eslint-plugin-cypress/pull/144) and that was released in [2.15.1](https://github.com/cypress-io/eslint-plugin-cypress/releases/tag/v2.15.1)
> * The author of PR [fix: unsafe-to-chain-command: Fix the false positive of 'focus' regex matching 'focused' #144](https://github.com/cypress-io/eslint-plugin-cypress/pull/144) @dominicfraser wrote in https://github.com/cypress-io/eslint-plugin-cypress/pull/141/files#r1329258120 however that he thought your solution was better.
> * Could you provide a test case which fails in [2.15.1](https://github.com/cypress-io/eslint-plugin-cypress/releases/tag/v2.15.1)  (or in the current version) and passes with your pending PR? I'm currently unsure what the exact issue is which this PR is intended to solve.

You can add the following rule to valid rules in `tests/lib/rules/unsafe-to-chain-command.js` and this is not going to work. The command is marked as unsafe to chain.

`{
      code: 'cy.submitBtn().click();',
      parserOptions,
},`

**Comment by MikeMcC399** [comment]
@gkorakas-eli 

Thank you for explaining and adding the extra test!

I found the reference in https://github.com/cypress-io/eslint-plugin-cypress/issues/140#issuecomment-1744326839 to the custom command `submitBtn` which I overlooked when I was trying to understand the history. Sorry for that.

**Comment by MikeMcC399** [comment]
@gkorakas-eli 

Please add the text

```markdown
- closes #140
```

to your original post so that this PR (#141) and the issue (#140) are automatically linked by GitHub.

**Review by MikeMcC399 — APPROVED** [review]
- Thanks for your submission! It passes the tests and fixes the issue from #140 🎉 

Tested against
https://github.com/cypress-io/cypress-example-kitchensink
using

```shell
npm install https://github.com/gkorakas-eli/eslint-plugin-cypress#fix/issue-with-unsafe-to-chain-command-that-is-not-action
```

and additional test-spec

```js
/// <reference types="cypress" />

context('eslinting', () => {
  it('custom command', () => {
    cy.submitBtn().click()
  })
})
```

| eslint-plugin-cypress | result     |
| --------------------- | ---------- |
| `2.13.4`              | no failure |
| `2.14.0`              | (1)(2)     |
| `2.15.1`              | (2)        |
| PR #141               | no failure |

(1) failure: `cy.focused().should('have.id', 'name')`
(2) failure: `cy.submitBtn().click()`

**Review by MikeMcC399 — APPROVED** [review]

**Review by jennifer-shehane — APPROVED** [review]

**Pull request merged**